doc/source/data/accessing-datasets.rst

```diff
@@ -4,47 +4,188 @@
 Accessing Datasets
 ===================
 
-Datasets can be passed to Ray tasks or actors and accessed with ``.iter_batches()`` or ``.iter_rows()``.
+The data underlying a ``Dataset`` can be accessed in several ways:
+
+* Retrieving a limited prefix of rows.
+* Iterating over rows and batches.
+* Converting into a Torch dataset or a TensorFlow dataset.
+* Converting into a RandomAccessDataset for random access.
+
+Retrieving limited set of rows
+==============================
+
+A limited set of rows can be retried from a ``Dataset`` via the
+:meth:`ds.take() <ray.data.Dataset.take>` API, along with its sibling helper APIs
+:meth:`ds.take_all() <ray.data.Dataset.take_all>`, for retrieving **all** rows, and
+:meth:`ds.show() <ray.data.Dataset.show>`, for printing a limited set of rows. These
+methods are convenient for quickly inspecting a subset (prefix) of rows. They have the
+benefit that, if used right after reading, they will only trigger more files to be
+read if needed to retrieve rows from that file; if inspecting a small prefix of rows,
+often only the first file will need to be read.
+
+.. literalinclude:: ./doc_code/accessing_datasets.py
+  :language: python
+  :start-after: __take_begin__
+  :end-before: __take_end__
+
+Iterating over Datasets
+=======================
+
+Datasets can be consumed a row at a time using the
+:meth:`ds.iter_rows() <ray.data.Dataset.iter_rows>` API
+
+.. literalinclude:: ./doc_code/accessing_datasets.py
+  :language: python
+  :start-after: __iter_rows_begin__
+  :end-before: __iter_rows_end__
+
+or a batch at a time using the
+:meth:`ds.iter_batches() <ray.data.Dataset.iter_batches>` API, where you can specify
+batch size as well as the desired batch format. By default, the batch format is
+``"native"``, which means that the batch format that's native to the data type will be
```

Comment:
Out of scope for this PR, but the native batch format has always been confusing to me. So is the batch format either Pandas Dataframe or a Python list? You recently added numpy batching format, how does that come into play? For each read API, what is the native format? Would be great to have this info somewhere in a future PR.

Reply:
The native batch format is a Pandas DataFrame for tabular data and a Python list for Python objects; we call this out in the `.map_batches()` docstring but agreed that we should have a dedicated section in one of the feature guides (or key concepts) about this.

>You recently added numpy batching format, how does that come into play?

That's not merged yet! Once it's merged, the native batch format for tensor data (represented as single-tensor-column Arrow tables) would be NumPy ndarrays.

>For each read API, what is the native format?

We _do_ have the block format for each read API documented in the [Creating Datasets feature guide](https://docs.ray.io/en/master/data/creating-datasets.html#creating-datasets), but we don't have the native batch format documented there.

This is a great call-out in general, I'll try to find time to open a follow-up PR for this today.

Reply:
The native batch format is a Pandas DataFrame for tabular data and a Python list for Python objects; we call this out in the `.map_batches()` docstring but agreed that we should have a dedicated section in one of the feature guides (or key concepts) about this.

>You recently added numpy batching format, how does that come into play?

That's not merged yet! Once it's merged, the native batch format for tensor data (represented as single-tensor-column Arrow tables) would be NumPy ndarrays.

>For each read API, what is the native format?

We _do_ have the block format for each read API documented in the [Creating Datasets feature guide](https://docs.ray.io/en/master/data/creating-datasets.html#creating-datasets), but we don't have the native batch format documented there.

This is a great call-out in general, I'll try to find time to open a follow-up PR for this today.

Reply:
I share the point that Amog had that the message is not quite clear: on one hand, Pandas is internal optimization; on the other it is so prominent that it's THE format for tabular, and it's exposed to users since it's used by UDF.

Reply:
Yep we should have a block format section in (I think) the key concepts page, that covers block representation and the native batch format, but I agreed with @amogkam that this should be out-of-scope for this PR.